ApplicationInsights/src/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -405,14 +424,14 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
 
         var configGetters: ITelemetryConfig = {
             instrumentationKey: () => config.instrumentationKey,
-            accountId: () => this.config.accountId,
-            sessionRenewalMs: () => this.config.sessionRenewalMs,
-            sessionExpirationMs: () => this.config.sessionExpirationMs,
-            sampleRate: () => this.config.samplingPercentage,
-            cookieDomain: () => this.config.cookieDomain,
-            sdkExtension: () => this.config.sdkExtension,
-            isBrowserLinkTrackingEnabled: () => this.config.isBrowserLinkTrackingEnabled,
-            appId: () => this.config.appId
+            accountId: () => this.config.accountId || config.accountId,
```

Comment:
One thing we could do is have a class implement IConfig. that can expose getConfig(key: string, extName: string) that returns the this.config[keyName] || this.config[extensionname][keyName]. Will make this look somewhat cleaner